### ISSUE TITLE
low correlation of gene count between HTSeq and RNA-SeQC

### ISSUE THREAD

**Issue by KangchengHou** [broadinstitute/gtex-pipeline#31]
Hi,

I am trying to adopt GTEx pipeline to replace the pipeline I have now. But when I try HTSeq and RNA-SeQC on the same set of BAM and annotation file. I find there is a difference between the two. 
For HTSeq (https://htseq.readthedocs.io/en/release_0.11.1/), I am using the following script
```
python3 -m HTSeq.scripts.count \
--stranded=no \
--mode=union \
        --type=gene ${BAM} ${GENEGTF} >> gene_union_count
```
And for RNA-SeQC v2(https://github.com/broadinstitute/rnaseqc), I am using the following script
```
${RNASEQC} ${GENE_GTF} ${BAM} ${OUT_DIR} --sample=${sample}
```
After I correlate the gene count output from two methods (${sample}.gene_reads.gct. I find very poor correlation (pearson correlation=~0.2, spearman correlation=~0.8). Also the coverage (sum of the gene count) is different (HTSeq ~10mil, RNA-SeQC ~6.5mil). HTSeq gives me the approximate coverage I expected.

Do you have idea on what's could be the possible cause for this difference? Thanks a lot in advance!

**Comment by agraubert** [comment]
Hello,

After looking at HTSeq's options, I have determined a few differences in operation which may account for the count difference you are observing.
1. HTSeq defaults to assuming the bam is sorted by read name, and requires `-r pos` to switch to position-sorted input. RNA-SeQC requires that the bam is sorted by position
2. HTSeq's mapping quality threshold is 10, whereas RNA-SeQC's is 255. The `-a {quality}` option to HTSeq or `-q {quality}` option to RNA-SeQC will change this behavior
3. HTSeq seems to have flags for controlling multi-mapping and supplemental alignments, however, I was not able to determine their default behavior from the README. RNA-SeQC ignores non-primary alignments and this behavior cannot be changed
4. I have noticed that the HTSeq counts are roughly half of RNA-SeQC. I have not seen a mention in the README, but that would indicate that HTSeq is reporting pairs instead of reads. Below is a figure of gene counts generated by the two tools
4. Lastly, RNA-SeQC requires a [collapsed gtf](https://github.com/broadinstitute/gtex-pipeline/tree/master/gene_model). This may cause significant changes in quantification, and also relates to the behavior of the `-m` flag for HTSeq.

When I ran both HTSeq and RNA-SeQC with a collapsed annotation, and with your given options, I obtained a pearson correlation of ~0.98 and a spearman correlation of ~0.88.

![download-1](https://user-images.githubusercontent.com/997749/72636643-7657d280-392d-11ea-9a30-47d5e5e017b3.png)
